### PR TITLE
fix #3122: support of PyYaml 5.1

### LINKED ISSUE #3122: Avoid using deprecated PyYAML API
Robotframework 3.1.1 works fine with PyYaml **3.13** on Python 3.7.2 Win64. Now switching to   new version of PyYaml **5.1**, deprecated YAMLLoadWarning is reported:

`robot\variables\filesetter.py:77: YAMLLoadWarning: calling yaml.load() without Loader=... is deprecated, as the default Loader is unsafe. Please read https://msg.pyyaml.org/load for full details.
  variables = yaml.load(stream)`

### PR DESCRIPTION
**Pull request by DirkRichter**
fix #3122 switch to PyYaml 5.1 FullLoader, which loads the full YAML language and avoids arbitrary code execution.

### CODE REVIEW

**Comment by pekkaklarck** [comment]
Is this backwards incompatible with older PyYaml versions? If yes, could the method to use be selected dynamically based on version?

**Comment by DirkRichter** [comment]
now it is backwards compatible with pyyaml v3.x (since 2006). Implementation is selected dynamically based on PyYaml version

File: src/robot/variables/filesetter.py [hunk 1]
```diff
@@ -74,7 +74,7 @@ def import_variables(self, path, args=None):
 
     def _import(self, path):
         with io.open(path, encoding='UTF-8') as stream:
-            variables = yaml.load(stream)
+            variables = yaml.load(stream) if yaml.__version__[0]=='3' else yaml.full_load(stream)  #pyyaml 3.x is from 2006
```
[the remark this answers is not in the document]
Comment (DirkRichter) [reply src/robot/variables/filesetter.py]: agree. this should be reverted

**Comment by pekkaklarck** [comment]
Do you @DirkRichter have time to update the PR? I think doc changes can be removed altogether and code could be changed like discussed above. I can fix this myself as well if you are busy.

**Comment by pekkaklarck** [comment]
Talked with @DirkRichter on Slack and he told he's too busy to get this PR cleaned up in the near future. We agreed I fix the issue myself and did that in ab9c9e4de0bdd4da21b472f12f23704d8c6126a4. 

Also submitted #3172 about updating PyYAML to v5.1 as this PR did.

**Pull request closed without merging**